src/dvc_data/index/index.py

```diff
@@ -442,6 +532,102 @@ def remote_exists(self, entry: "DataIndexEntry", **kwargs) -> bool:
 
         return storage.remote.exists(entry, **kwargs)
 
+    def _bulk_storage_exists(
+        self,
+        entries: list[DataIndexEntry],
+        storage_selector: Callable[["StorageInfo"], Optional["Storage"]],
```

Comment:
```suggestion
        storage: str = "cache"
```

I'd use `getattr(storage_info, storage)` to get the actual `storage_obj` similar to how we do in:

https://github.com/treeverse/dvc-data/blob/ba5e2c4e37e051a40e5905e25f821cd8b6c126ea/src/dvc_data/index/checkout.py#L78
https://github.com/treeverse/dvc-data/blob/ba5e2c4e37e051a40e5905e25f821cd8b6c126ea/src/dvc_data/index/checkout.py#L90

Reply:
The `bulk_remote_exists` and `bulk_cache_exists` can then just pass stringified values.

Reply:
Agreed, that seems simpler. Changed it accordingly.